### ISSUE TITLE
[BUG] Salt 3006.2 relenv sets OPENSSL_MODULES, which breaks certbot when invoked via cmd.run

### ISSUE THREAD

**Issue by higels** [saltstack/salt#65058]
**Description**
On an OpenSSL3 based system with salt 3006.2, calling certbot via cmd.run results in a lengthy stack trace due to `cryptography.exceptions.InternalError: Unknown OpenSSL error`. Changing the cmd.run argument to `unset OPENSSL_MODULES; certbot` works as expected.

**Setup**

Please be as specific as possible and give set-up details.

- [x] on-prem machine
- [ ] VM (Virtualbox, KVM, etc. please specify)
- [ ] VM running on a cloud service, please be explicit and add details
- [ ] container (Kubernetes, Docker, containerd, etc. please specify)
- [ ] or a combination, please be explicit
- [ ] jails if it is FreeBSD
- [ ] classic packaging
- [x] onedir packaging
- [ ] used bootstrap to install

Ubuntu 22.04.2 LTS
system python 3.10.12-1~22.04.2
python3-cryptography 3.4.8-1ubuntu2
certbot 1.21.0-1build1
Salt is 3006.2 from repo.saltproject.io/salt/py3/ubuntu/22.04/amd64/3006

**Steps to Reproduce the behavior**
Compare and contrast `salt-call cmd.run 'certbot'` to `salt-call cmd.run 'unset OPENSSL_MODULES; certbot'`

**Expected behavior**
Certbot should run normally without an unhandled stack trace.

**Screenshots**
Exact output is:
```
local:
    Traceback (most recent call last):
      File "/usr/bin/certbot", line 33, in <module>
        sys.exit(load_entry_point('certbot==1.21.0', 'console_scripts', 'certbot')())
      File "/usr/bin/certbot", line 25, in importlib_load_entry_point
        return next(matches).load()
      File "/usr/lib/python3.10/importlib/metadata/__init__.py", line 171, in load
        module = import_module(match.group('module'))
      File "/usr/lib/python3.10/importlib/__init__.py", line 126, in import_module
        return _bootstrap._gcd_import(name[level:], package, level)
      File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
      File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
      File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
      File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
      File "<frozen importlib._bootstrap_external>", line 883, in exec_module
      File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
      File "/usr/lib/python3/dist-packages/certbot/main.py", line 2, in <module>
        from certbot._internal import main as internal_main
      File "/usr/lib/python3/dist-packages/certbot/_internal/main.py", line 17, in <module>
        import josepy as jose
      File "/usr/lib/python3/dist-packages/josepy/__init__.py", line 43, in <module>
        from josepy.json_util import (
      File "/usr/lib/python3/dist-packages/josepy/json_util.py", line 14, in <module>
        from OpenSSL import crypto
      File "/usr/lib/python3/dist-packages/OpenSSL/__init__.py", line 8, in <module>
        from OpenSSL import crypto, SSL
      File "/usr/lib/python3/dist-packages/OpenSSL/crypto.py", line 17, in <module>
        from OpenSSL._util import (
      File "/usr/lib/python3/dist-packages/OpenSSL/_util.py", line 6, in <module>
        from cryptography.hazmat.bindings.openssl.binding import Binding
      File "/usr/lib/python3/dist-packages/cryptography/hazmat/bindings/openssl/binding.py", line 189, in <module>
        Binding.init_static_locks()
      File "/usr/lib/python3/dist-packages/cryptography/hazmat/bindings/openssl/binding.py", line 163, in init_static_locks
        cls._ensure_ffi_initialized()
      File "/usr/lib/python3/dist-packages/cryptography/hazmat/bindings/openssl/binding.py", line 151, in _ensure_ffi_initialized
        _openssl_assert(
      File "/usr/lib/python3/dist-packages/cryptography/hazmat/bindings/openssl/binding.py", line 77, in _openssl_assert
        raise InternalError(
    cryptography.exceptions.InternalError: Unknown OpenSSL error. This error is commonly encountered when another library is not cleaning up the OpenSSL error stack. If you are using cryptography with another library that uses OpenSSL try disabling it before reporting a bug. Otherwise please file an issue at https://github.com/pyca/cryptography/issues with information on how to reproduce this. ([_OpenSSLErrorWithText(code=310378599, lib=37, reason=103, reason_text=b'error:12800067:DSO support routines::could not load the shared library'), _OpenSSLErrorWithText(code=310378599, lib=37, reason=103, reason_text=b'error:12800067:DSO support routines::could not load the shared library'), _OpenSSLErrorWithText(code=126615813, lib=15, reason=786693, reason_text=b'error:078C0105:common libcrypto routines::init fail')])
```

**Versions Report**
<details><summary>salt --versions-report</summary>
(Provided by running salt --versions-report. Please also mention any differences in master/minion versions.)

```yaml
Salt Version:
          Salt: 3006.2

Python Version:
        Python: 3.10.12 (main, Aug  3 2023, 21:47:10) [GCC 11.2.0]

Dependency Versions:
          cffi: 1.14.6
      cherrypy: 18.6.1
      dateutil: 2.8.1
     docker-py: Not Installed
         gitdb: Not Installed
     gitpython: Not Installed
        Jinja2: 3.1.2
       libgit2: Not Installed
  looseversion: 1.0.2
      M2Crypto: Not Installed
          Mako: Not Installed
       msgpack: 1.0.2
  msgpack-pure: Not Installed
  mysql-python: Not Installed
     packaging: 22.0
     pycparser: 2.21
      pycrypto: Not Installed
  pycryptodome: 3.9.8
        pygit2: Not Installed
  python-gnupg: 0.4.8
        PyYAML: 6.0.1
         PyZMQ: 23.2.0
        relenv: 0.13.3
         smmap: Not Installed
       timelib: 0.2.4
       Tornado: 4.5.3
           ZMQ: 4.3.4

System Versions:
          dist: ubuntu 22.04.2 jammy
        locale: utf-8
       machine: x86_64
       release: 5.15.0-79-generic
        system: Linux
       version: Ubuntu 22.04.2 jammy
```
</details>

**Additional context**
After identifying the root cause, we fixed our salt states but it is likely that any external python based program that relies on python3-cryptography is going to break similarly to this on system which uses OpenSSL3. I reproduced this with mitmproxy as well.

It appears that relenv is responsible for setting the OPENSSL_MODULES directory. Perhaps salt could make a copy of the environment before relenv touches it and explicitly pass that down to cmd.run and friends. This seems like a safe long term approach.

**Comment by dwoz** [comment]
This is also being tracked here: https://github.com/saltstack/relative-environment-for-python/issues/140

**Comment by 9967819** [comment]
This is a cryptography specific module error. I have found this problem after upgrading to python 3.11. As a workaround i simply deleted the python3-cryptography and reinstalled cryptography module using pip in user mode using --break-system-packages option. Cheers.

**Comment by dwoz** [comment]
This has been resolved in `3006.3`
